lib/llvmopencl/ImplicitLoopBarriers.cc

```diff
@@ -116,8 +116,13 @@ static bool addInnerLoopBarrier(llvm::Loop &L,
 
     /* Add a barrier both to the beginning of the entry and to the very end
        to nicely isolate the parallel region. */
+#if LLVM_MAJOR < 20
     Barrier::create(brexit->getTerminator());
     Barrier::create(loopEntry->getFirstNonPHI());
+#else
+    Barrier::create(brexit->getTerminator()->getIterator());
```

Comment:
Looking at these. Why not move the getIterator() call inside Barrier::create and avoid a lot of these ifdefs?

Reply:
fixed the ifdefs